pkg/cloudprovider/instance.go

```diff
@@ -56,24 +57,28 @@ var (
 )
 
 type InstanceProvider struct {
-	region                 string
-	ec2api                 ec2iface.EC2API
-	unavailableOfferings   *cache.UnavailableOfferings
-	instanceTypeProvider   *InstanceTypeProvider
-	subnetProvider         *subnet.Provider
-	launchTemplateProvider *LaunchTemplateProvider
-	createFleetBatcher     *CreateFleetBatcher
+	region                    string
+	ec2api                    ec2iface.EC2API
+	unavailableOfferings      *cache.UnavailableOfferings
+	instanceTypeProvider      *InstanceTypeProvider
+	subnetProvider            *subnet.Provider
+	launchTemplateProvider    *LaunchTemplateProvider
+	createFleetBatcher        *batcher.CreateFleetBatcher
+	terminateInstancesBatcher *batcher.TerminateInstancesBatcher
+	describeInstancesBatcher  *batcher.DescribeInstancesBatcher
 }
 
 func NewInstanceProvider(ctx context.Context, region string, ec2api ec2iface.EC2API, unavailableOfferings *cache.UnavailableOfferings, instanceTypeProvider *InstanceTypeProvider, subnetProvider *subnet.Provider, launchTemplateProvider *LaunchTemplateProvider) *InstanceProvider {
 	return &InstanceProvider{
-		region:                 region,
-		ec2api:                 ec2api,
-		unavailableOfferings:   unavailableOfferings,
-		instanceTypeProvider:   instanceTypeProvider,
-		subnetProvider:         subnetProvider,
-		launchTemplateProvider: launchTemplateProvider,
-		createFleetBatcher:     NewCreateFleetBatcher(ctx, ec2api),
+		region:                    region,
+		ec2api:                    ec2api,
+		unavailableOfferings:      unavailableOfferings,
+		instanceTypeProvider:      instanceTypeProvider,
+		subnetProvider:            subnetProvider,
+		launchTemplateProvider:    launchTemplateProvider,
+		createFleetBatcher:        batcher.NewCreateFleetBatcher(ctx, ec2api),
+		terminateInstancesBatcher: batcher.NewTerminateInstancesBatcher(ctx, ec2api),
+		describeInstancesBatcher:  batcher.NewDescribeInstancesBatcher(ctx, ec2api),
```

Comment:
Optional: Use the same pattern we use in our fakes to wrap the ec2api with batch implementations.

`ec2api: batch.EC2API(ec2api)`

Reply:
wdyt about (updated):

```
ec2Batcher:             batcher.EC2(ctx, ec2api),
```

Reply:
The idea was to abstract the fact that our EC2 client was a batching client. Whether it's batch or batcher, I'm indifferent, but I think `batch` is marginally cleaner semantically.

Reply:
the problem with completely abstracting that the EC2 client is a batching client is that they have to individually be implemented since batching means different things for different APIs.  So I still like the idea that they're separate clients since they do things functionally different. We may also want non-batching versions of those calls too for a specific purpose.